src/test/common/clara/test_simple_rules.cljc

```diff
@@ -194,3 +194,14 @@
     (is (has-fact? (:cold @side-effect-holder) (->Temperature -10 "MCI") ))
 
     (is (has-fact? (:subzero @side-effect-holder) (->Temperature -10 "MCI")))))
+
+(def-rules-test test-condition-with-empty-constraint
+  {:rules [simple-rule [[[Cold () (not= this ())]]
```

Comment:
This is good, but looking over compile-constraints I think we should have a test where we validate that a rule with only empty constraints, preferably for both the case of 1 empty constraint and 2 empty constraints, works correctly.  The case of no constraints (also a recursion termination case here) is specifically handled with this:

````
(if (empty? exp-seq)
     `(deref ~'?__bindings__)
````

but I'd like to make sure that we're testing the recursion properly both when there are later non-empty constraints and when we bottom out without finding any meaningful constraints.